### PR TITLE
can: fix RTR ioctl and support timeout

### PR DESCRIPTION
**Pull request by MTres19**
## Summary
The RTR ioctl was not really usable (and few (any?) drivers implemented it. However, it's potentially useful and doesn't require a lot of work to support.

## Impact
No changes needed to existing drivers. If drivers would like to support this ioctl, they should support/enable `CONFIG_CAN_USE_RTR` and set `co_remoterequest` to NULL.

## Testing
Tested with TM4C123G Launchpad. [Test application](https://github.com/MTres19/ETCetera-tools/blob/dab39c7981c71715266e2fc69cc30c7aea86ef10/cantest_main.c#L556).

### CODE REVIEW

File: drivers/can/can.c [hunk 1]
```diff
@@ -916,12 +918,15 @@ static ssize_t can_write(FAR struct file *filep, FAR const char *buffer,
...
  ****************************************************************************/
 
-static inline ssize_t can_rtrread(FAR struct can_dev_s *dev,
-                                  FAR struct canioc_rtr_s *rtr)
+static inline ssize_t can_rtrread(FAR struct file *filep,
+                                  FAR struct canioc_rtr_s *request)
 {
-  FAR struct can_rtrwait_s *wait = NULL;
+  FAR struct can_dev_s *dev       = filep->f_inode->i_private;
+  FAR struct can_rtrwait_s *wait  = NULL;
```
Comment (xiaoxiang781216) [inline drivers/can/can.c:925]: remvoe the extra space before =

Comment (MTres19) [reply drivers/can/can.c:925]: Fixed now

Comment (xiaoxiang781216) [reply drivers/can/can.c:925]: but @MTres19 I mean wait variable too

Comment (MTres19) [reply drivers/can/can.c]: Whoops, this time it's really fixed I think @xiaoxiang781216. Thanks.

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**